# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../openoffice-org/patches/xulrunner-1.9.diff
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- config_office/configure.in.orig	2008-01-31 05:45:01.000000000 -0600
+++ config_office/configure.in	2008-03-29 14:19:38.000000000 -0500
@@ -466,7 +466,8 @@
                           components cannot be built against a contemporary
                           mozilla. The flavour used can be specified by
                           --with-system-mozilla=<flavour>. Supported are:
-                          xulrunner (default), firefox, seamonkey, mozilla
+                          libxul (default), xulrunner, firefox, seamonkey,
+                          mozilla
 ], WITH_SYSTEM_MOZILLA=$withval, WITH_SYSTEM_MOZILLA=no)
 AC_ARG_WITH(stlport,
 [  --with-stlport         The location that STLport is installed in. The STL
@@ -3781,7 +3782,9 @@
     AC_MSG_RESULT([external])
     SYSTEM_MOZILLA=YES
     AC_MSG_CHECKING([which Mozilla flavour to use])
-    if test -n "$with_system_mozilla" && test "$with_system_mozilla" = "xulrunner"; then
+    if test -n "$with_system_mozilla" && test "$with_system_mozilla" = "libxul"; then
+      MOZ_FLAVOUR=libxul
+    elif test -n "$with_system_mozilla" && test "$with_system_mozilla" = "xulrunner"; then
       MOZ_FLAVOUR=xulrunner
     elif test -n "$with_system_mozilla" && test "$with_system_mozilla" = "seamonkey"; then
       MOZ_FLAVOUR=seamonkey
@@ -3790,12 +3793,18 @@
     elif test -n "$with_system_mozilla" && test "$with_system_mozilla" = "mozilla"; then
       MOZ_FLAVOUR=mozilla
     else
-      MOZ_FLAVOUR=xulrunner
+      MOZ_FLAVOUR=libxul
     fi
     tmp=`echo $MOZ_FLAVOUR | $PERL -e 'print ucfirst(<STDIN>);'`
     AC_MSG_RESULT($tmp)
 
-    PKG_CHECK_MODULES( MOZILLAXPCOM, $MOZ_FLAVOUR-xpcom )
+    PKG_CHECK_MODULES( MOZ_NSS, nss, STANDALONENSS="TRUE", STANDALONENSS="" )
+    if test -z "$STANDALONENSS"; then
+        PKG_CHECK_MODULES( MOZ_NSS, $MOZ_FLAVOUR-nss )
+    else
+        NSS_LIB="-L`$PKG_CONFIG --variable=libdir nss`"
+        AC_SUBST(NSS_LIB)
+    fi
 
     if $PKG_CONFIG --exists nspr ; then
       PKG_CHECK_MODULES( MOZ_NSPR, nspr )
@@ -3805,12 +3814,20 @@
       PKG_CHECK_MODULES( MOZ_NSPR, $MOZ_FLAVOUR-nspr )
     fi
 
-    PKG_CHECK_MODULES( MOZ_NSS, nss, STANDALONENSS="TRUE", STANDALONENSS="" )
-    if test -z "$STANDALONENSS"; then
-        PKG_CHECK_MODULES( MOZ_NSS, $MOZ_FLAVOUR-nss )
+    if $PKG_CONFIG --exists libxul ; then
+      PKG_CHECK_MODULES( MOZILLAXPCOM, libxul )
+      MOZ_INC=`$PKG_CONFIG --variable=includedir libxul`
+      MOZ_LIB=`$PKG_CONFIG --variable=libdir libxul`
+      if ! test -e "$MOZ_LIB/libxul.so"; then
+         MOZ_LIB=`$PKG_CONFIG --variable=sdkdir libxul`
+         if test -e "$MOZ_LIB/sdk/lib/libxul.so"; then
+            MOZ_LIB="$MOZ_LIB/sdk/lib"
+         fi
+      fi
     else
-        NSS_LIB="-L`$PKG_CONFIG --variable=libdir nss`"
-        AC_SUBST(NSS_LIB)
+      PKG_CHECK_MODULES( MOZILLAXPCOM, $MOZ_FLAVOUR-xpcom )
+      MOZ_INC=`$PKG_CONFIG --variable=includedir $MOZ_FLAVOUR-xpcom`
+      MOZ_LIB=`$PKG_CONFIG --variable=libdir $MOZ_FLAVOUR-xpcom`
     fi
 
     save_CPPFLAGS="$CPPFLAGS"
@@ -3826,8 +3843,6 @@
     CPPFLAGS="$save_CPPFLAGS"
     LIBS="$save_LIBS"
 
-    MOZ_INC=`$PKG_CONFIG --variable=includedir $MOZ_FLAVOUR-xpcom`
-    MOZ_LIB=`$PKG_CONFIG --variable=libdir $MOZ_FLAVOUR-xpcom`
     MOZ_LIB_XPCOM=$MOZILLAXPCOM_LIBS
     if test "$WITH_LDAP" != "NO" && test "$WITH_OPENLDAP" != "YES"; then
       AC_MSG_CHECKING([whether $tmp was compiled with --enable-ldap])
@@ -3840,6 +3855,13 @@
 Please recompile $tmp with --enable-ldap or use --with-openldap.])
       fi
     fi
+
+    #e.g. http://fedoraproject.org/wiki/Releases/FeatureXULRunnerAPIChanges
+    #the plugin pkg-config etc. reverts to "mozilla-plugin" with libxul
+    if test "$MOZ_FLAVOUR" == "libxul"; then
+        MOZ_FLAVOUR="mozilla"
+    fi
+
 elif test "$enable_mozilla" = "no"; then
     AC_MSG_RESULT([none])
     WITH_MOZILLA=NO
